proposals/testing/NNNN-evaluate-condition.md

```diff
@@ -0,0 +1,74 @@
+# Public API to evaluate ConditionTrait
+
+* Proposal: [SWT-NNNN](NNNN-evaluate-condition.md)
+* Authors: [David Catmull](https://github.com/Uncommon)
+* Status: **Awaiting review**
+* Implementation: [swiftlang/swift-testing#909](https://github.com/swiftlang/swift-testing/pull/909)
+* Review: ([pitch](https://forums.swift.org/t/pitch-introduce-conditiontrait-evaluate/77242))
+
+## Introduction
+
+This adds an `evaluate()` method to `ConditionTrait` to evaluate the condition
+without requiring a `Test` instance.
+
+## Motivation
+
+Currently, the only way a `ConditionTrait` is evaluated is inside the
+`prepare(for:)` method. This makes it difficult for third-party libraries to
+utilize these traits because evaluating a condition would require creating a
+dummy `Test` to pass to that method.
+
+## Proposed solution
+
+The proposal is to add a `ConditionTrait.evaluate()` method which returns the
+result of the evaluation. The existing `prepare(for:)` method is updated to call
+`evaluate()` so that the logic is not duplicated.
+
+## Detailed design
+
+The `evaluate()` method is as follows, containing essentially the same logic
+as was in `prepare(for:)`:
+
+```swift
+public func evaluate() async throws -> EvaluationResult {
```

Comment:
Please show this in an extension of `ConditionTrait` to make it clear it's a member, and include markup documentation. :)

Reply:
Done